packages/clay-date-picker/src/index.tsx

```diff
@@ -591,12 +597,13 @@ const ClayDatePicker = React.forwardRef<HTMLInputElement, IProps>(
 		return (
 			<FocusScope>
 				<div className="date-picker">
-					<ClayInput.Group id={id} ref={triggerElementRef}>
+					<ClayInput.Group id={inputGroupId} ref={triggerElementRef}>
```

Comment:
@pat270 these ids for association I believe we can provide this via OOTB, for example see the `useId()` hook in the VerticalBar component, it will generate a unique incremental id this is better because it is more difficult for DXP developers to deal with the generation of ids and normally they don't add.

Reply:
@matuzalemsteles I don't think we need an `id` on `input-group`. I just added `inputGroupId` because we were previously allowed to add an `id` there. I tried `useId()` but it generates it on every change which isn't useful when trying to target it with CSS. I'm going to remove it.

Reply:
The use case in portal is:

```html
<label for="idGeneratedByPortal"></label>
<ClayDatePicker id="idGeneratedByPortal" />
```

If we generate for them, they won't be able to pass their custom id.

Reply:
On second thought I hadn't seen your change, but you're right it doesn't make sense to associate the group with the input just in that case, maybe just leaving the recent change as it is is enough. We can also use `useId` at any time to associate things.

Looking at this component it looks like we have some issues with the menu trigger with the menu, it doesn't correctly associate if the menu is open or not, I think we need to fix that too.